graphstorm-processing/docs/source/developer/developer-guide.rst

```diff
@@ -0,0 +1,227 @@
+Developer Guide
+---------------
+
+The project is set up using ``poetry`` to make easier for developers to
+jump into the project.
+
+The steps we recommend are:
+
+Install JDK 8, 11 or 17
+~~~~~~~~~~~~~~~~~~~~~~~
+
+PySpark requires a compatible Java installation to run, so
+you will need to ensure your active JDK is using either
+Java 8 or 11.
+
+On MacOS you can do this using ``brew``:
```

Comment:
I think we can support it, since Spark runs on the JVM, once the environment is properly configured the OS doesn't really matter.

I know PyTest works for MacOS, I'll make sure local execution works as well.

Reply:
Verified the example works on MacOS on fresh clone, resolving.